cloudferrylib/os/identity/keystone.py

```diff
@@ -269,6 +268,15 @@ def func(tenant_id):
 
         return func
 
+    def get_tenants_func_without_admin(self):
```

Comment:
I would suggest to modify `get_tenants_func` to have additional argument, called `default_tenant` which is set to `config.cloud.tenant` by default, e.g.:
```
def get_tenants_func(self, default_tenant=None):
    tenants = {tenant.id: tenant.name
               for tenant in self.get_tenants_list()}

    def func(tenant_id):
        if default_tenant is None:
            default_tenant = self.config.cloud.tenant
        return tenants.get(tenant_id, default_tenant)

    return func
```
This will resolve:
 - hardcoded `'admin'` value related issues (e.g. if cloud does not have 'admin' tenant defined at all)
 - reduce number of redundant methods